Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al

```diff
@@ -525,6 +537,115 @@ codeunit 4022 "GP Vendor Migrator"
+
+            // Additional Vendor accounts
+            GPPM00203.SetRange(VENDORID, VendorNo);
+            if GPPM00203.FindSet() then
+                repeat
+                    if GPPM00203.ACTINDX > 0 then
+                        CreateRecurringLineFromAccount(VendorNo, GPPM00203.ACTINDX);
+                until GPPM00203.Next() = 0;
+
+            exit;
+        end;
+
+        // Vendor class level
+        if GPCompanyAdditionalSettings.GetMigrateVendorClasses() then
+            if GPPM00100.Get(GPVendor.VNDCLSID) then
+                if GPPM00100.PMPRCHIX > 0 then begin
+                    CreateRecurringLineFromAccount(VendorNo, GPPM00100.PMPRCHIX);
+
+                    // Additional Vendor class accounts
+                    GPPM00101.SetRange(VNDCLSID, GPVendor.VNDCLSID);
+                    if GPPM00101.FindSet() then
+                        repeat
+                            if GPPM00101.ACTINDX > 0 then
+                                CreateRecurringLineFromAccount(VendorNo, GPPM00101.ACTINDX);
+                        until GPPM00101.Next() = 0;
+                    exit;
+                end;
+
+        // Fallback - Purchase Account
+        if GPPostingAccounts.FindFirst() then
+            if GPPostingAccounts.PurchAccountIdx > 0 then
+                CreateRecurringLineFromAccount(VendorNo, GPPostingAccounts.PurchAccountIdx);
+    end;
+
+    local procedure CreateRecurringLineFromAccount(VendorNo: Code[20]; GPActIdx: Integer)
+    var
+        GPAccount: Record "GP Account";
+        GLAccount: Record "G/L Account";
+    begin
+        if not GPAccount.Get(GPActIdx) then
+            exit;
+
+        if not GLAccount.Get(GPAccount.AcctNum) then
+            exit;
+
+        CreateRecurringPurchaseLineImp(VendorNo, GPAccount);
+    end;
+
+    local procedure CreateRecurringPurchaseLineImp(VendorNo: Code[20]; var GPAccount: Record "GP Account")
+    var
+        StandardPurchaseCode: Record "Standard Purchase Code";
+        StandardPurchaseLine: Record "Standard Purchase Line";
+        StandardVendorPurchaseCode: Record "Standard Vendor Purchase Code";
+        NoSeries: Codeunit "No. Series";
+        HelperFunctions: Codeunit "Helper Functions";
+        DimSetID: Integer;
+    begin
+        if GPAccount."Standard Purchase Code" = '' then begin
+            GPAccount."Standard Purchase Code" := CopyStr(NoSeries.GetNextNo(NoSeriesStandardPurchaseCodeTok), 1, MaxStrLen(GPAccount."Standard Purchase Code"));
+
+            StandardPurchaseCode.Validate("Code", GPAccount."Standard Purchase Code");
+            StandardPurchaseCode.Validate(Description, CopyStr(HelperFunctions.GenerateStandardCodeDescriptionFromAccount(GPAccount), 1, MaxStrLen(StandardPurchaseCode.Description)));
+            StandardPurchaseCode.Insert(true);
+
+            StandardPurchaseLine.Validate("Standard Purchase Code", StandardPurchaseCode."Code");
+            StandardPurchaseLine.Validate("Line No.", 1);
```

Comment:
Same comment we increment by 10000. Consider applying same comments as for customer.